[Bryan Ng] iP

bryanngzh opened the pull request:
## Duke
> "As much as this world hurt me...it's mine...and I will kill to defend it." – Reyna ([source](https://valorant.fandom.com/wiki/Reyna/Quotes))

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Adding Notes (coming soon)
- [ ] Reminders (coming soon)
--- 
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

bryanngzh committed:
Level 1 (Greet, Echo, Exit)

bryanngzh committed:
Level-2: add, list

bryanngzh committed:
Level-3: mark tasks as done

bryanngzh committed:
A-Classes: added Task class

bryanngzh committed:
A-Inheritance: added Deadline, Event, Todo classes

bryanngzh committed:
Level-4: add Todo, Deadline and Event commands

bryanngzh committed:
A-TextUiTesting: Automated Text UI Testing

bryanngzh committed:
A-Exceptions: Added DukeException class

bryanngzh committed:
Level-5: Handle Errors for Duke

bryanngzh committed:
A-Collections: using ArrayList to store Task

bryanngzh committed:
Level-6: Added delete task functionality

bryanngzh committed:
Fixed DukeException error message

bryanngzh committed:
A-Enums: Added usage of Enums to Duke

bryanngzh committed:
Fixed automated test cases

bryanngzh committed:
Added more automated test cases

bryanngzh committed:
Level-7: Save data in file

bryanngzh committed:
Level-8: Add dates and times

bryanngzh committed:
Merge branch 'branch-Level-7'
Merge branch-Level-7 to master branch

bryanngzh committed:
Merge branch 'branch-Level-8'

bryanngzh committed:
Merge branch 'add-gradle-support'

bryanngzh committed:
Update build.gradle

bryanngzh committed:
A-MoreOOP: refactor code with more OOP

bryanngzh committed:
A-Packages: Divide classes into packages

bryanngzh committed:
A-JUnit: Add JUnit tests

bryanngzh committed:
A-JavaDoc: Add JavaDoc comments for classes and methods

bryanngzh committed:
A-CodingStandard: Fix code style violations

bryanngzh committed:
Level-9: Add a way to find task by searching for keyword

bryanngzh committed:
Merge branch 'branch-A-JavaDoc'

bryanngzh committed:
Merge branch 'branch-A-CodingStandard'

bryanngzh committed:
Merge branch 'branch-Level-9'

bryanngzh committed:
Add checkstyle requirements to code

bryanngzh committed:
A-CheckStyle: Add checkstyle for code

bryanngzh committed:
Merge branch 'branch-A-CheckStyle'

guokweijie commented on src/main/java/duke/command/OnDateCommand.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import duke.exception.DukeException;
```
good separation of classes into the different packages!

guokweijie commented on src/main/java/duke/command/OnDateCommand.java:
```diff
@@ -0,0 +1,63 @@
+    @Override
+    public void execute(Ui ui, Storage storage, TaskList taskList) throws DukeException {
+        String formattedDate = date.format(DateTimeFormatter.ofPattern("E, d MMM yyyy"));
+        ArrayList<Task> filteredTasks = taskList.getTasks().stream().filter(task -> task.isOn(date))
```
good application of streams from CS2030S!

guokweijie commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+            secondWord = inputArray[1];
+        }
+        switch (firstWord) {
+        case "bye":
```
good indentation of switch and case!

guokweijie commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+                if (secondWord.length() == 0) {
+                    throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+                }
+                String[] dArray = secondWord.split(" /by ", 2);
```
name can be more descriptive

bryanngzh replied:
Thanks for the feedback, I will update this in the future!

guokweijie reviewed (commented):
Overall your code is pretty good and follows coding standard and has good code quality. keep up the good work!

jehousoh commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,49 @@
+            ui.formatMessage(String.format("Now you have %d tasks in the list.", taskList.size()));
+        }
+    }
+
```
Should there be a javadoc here?

jehousoh commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+     *
+     * @return a boolean value.
+     */
+    public abstract boolean isRunning();
```
Should you implement a return true here and overwrite it with a return false in the ExitCommand class instead?

bryanngzh replied:
Sounds like a great idea, thanks!

jehousoh commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+    }
+
+    /**
+     * Constructor for TaskList.
```
Could you be more specific what the difference between this and the constructor above is?

jehousoh commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,61 @@
+     */
+    @Override
+    public void execute(Ui ui, Storage storage, TaskList taskList) throws DukeException {
+        ArrayList<Task> filteredTasks = taskList.getTasks().stream().filter(task -> task.contains(keyword))
```
Could you use line breaks to improve readability?

jehousoh reviewed (commented):
Looks good! 👍 
Just some minor suggestions for changes, might want to consider line breaks to improve some readability!
Jiayous and keep up the ~~good~~ awesome work!!

jehousoh commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,49 @@
+        }
+    }
+
+    @Override
```
Should there be a javadoc here

bryanngzh replied:
Will update this in the future!

bryanngzh committed:
Level-10: Add GUI to duke

bryanngzh committed:
Merge branch 'branch-Level-10'

bryanngzh committed:
Fix empty description bug for todo

bryanngzh committed:
Fix checkstyle issues pointed out by reviewers

Checkstyle issues present in MainWindow, AddCommand and TaskList.

This caused checkstyle to fail and led to a messy structure of code.

By adjusting the indentation and ordering of import statements for MainWindow
to the correct order, checkstyle succeded.

I missed out some comments in AddCommand and TaskList as well.

By adding the appropriate comments, the code became clearer and easier to read.

bryanngzh committed:
A-Assertions: Add assertions to commands

The assertions help to verify that Duke is sending out messages
whenever a command is run.
This helps to ensure that the Duke bot is working fine and is
replying the user.

bryanngzh committed:
A-CodeQuality: Minimize code duplication and magic numbers

Error messages in Parser were often the same message, which resulted in duplication of code.
By creating a constant to refer to, magic numbers and code duplication is reduced.

bryanngzh committed:
A-Streams: Add Streams into FindCommand

bryanngzh committed:
Merge pull request #3 from bryanngzh/branch-A-CodeQuality

A-CodeQuality: Minimize code duplication and magic numbers

bryanngzh committed:
Merge branch 'master' into branch-A-Assertions

bryanngzh committed:
Merge branch 'master' into branch-A-Streams

bryanngzh committed:
Merge pull request #2 from bryanngzh/branch-A-Assertions

A-Assertions: Add assertions to commands

bryanngzh committed:
Merge branch 'master' into branch-A-Streams

bryanngzh committed:
Merge pull request #1 from bryanngzh/branch-A-Streams

A-Streams: Add streams to TaskList

bryanngzh committed:
A-CI: Add gradle file

bryanngzh committed:
B-ViewSchedules: Add on command to see tasks on that date

bryanngzh committed:
Merge branch 'branch-B-ViewSchedules'

bryanngzh committed:
Code Quality: Reduce method length in parser

bryanngzh committed:
Improve GUI

bryanngzh committed:
Update the bot's personality to Shin Yuna

bryanngzh committed:
Fix bug related to gradle

bryanngzh committed:
Fix checkstyle

bryanngzh committed:
Add a user guide

bryanngzh committed:
Fix bug where GUI does not close after bye

bryanngzh committed:
Add Ui.png

bryanngzh committed:
Fix error message for mark -100 command

bryanngzh committed:
Tasks save to storage only after Add, Delete, Mark and Unmark

bryanngzh committed:
Change image type to png

bryanngzh committed:
Update README.md

bryanngzh committed:
Update README.md
